pkg/volume/empty_dir/empty_dir.go

```diff
@@ -66,27 +66,29 @@ func (plugin *emptyDirPlugin) Name() string {
 	return emptyDirPluginName
 }
 
-func (plugin *emptyDirPlugin) CanSupport(spec *api.Volume) bool {
+func (plugin *emptyDirPlugin) CanSupport(spec *volume.InternalVolume) bool {
 	if plugin.legacyMode {
 		// Legacy mode instances can be cleaned up but not created anew.
 		return false
 	}
 
-	if util.AllPtrFieldsNil(&spec.VolumeSource) {
+	if util.AllPtrFieldsNil(spec.PersistentVolumeSource) ||
```

Comment:
This is a bit weird, actually.  It means that EmptyDir will trigger in the case of a malformed PVS, which is NOT what we want.

Perhaps we should make the internal struct hold pointers to PVS and VS, and that way EmptyDir would only consider if spec.VS != nil && AllPtrFieldsNil(spec.VS)

What think?